src/server.rs

```diff
@@ -1049,6 +1049,22 @@ impl<B: Buf> SendResponse<B> {
             .map_err(Into::into)
     }
 
+    /// Send a non-final 1xx response to a client request.
+    ///
+    /// The [`SendResponse`] instance is already associated with a received
+    /// request.  This function may only be called if [`send_reset`] or
+    /// [`send_response`] has not been previously called.
+    ///
+    /// [`SendResponse`]: #
+    /// [`send_reset`]: #method.send_reset
+    /// [`send_response`]: #method.send_response
+    pub fn send_info(&mut self, response: Response<()>) -> Result<(), crate::Error> {
```

Comment:
Perhaps this just needs to document with a `# Panics` section that passing a response with a non-1xx status code is a misuse of the method.

Reply:
Agreed, and fixed.